### PR TITLE
rbd-mirror: keep events from different epochs independent

### PR DESCRIPTION
**Pull request by dillaman**
Fixes: http://tracker.ceph.com/issues/16362
Signed-off-by: Jason Dillaman <dillaman@redhat.com>

### CODE REVIEW

**Comment by trociny** [comment]
A note for merging: it will need rebase after #9788 merge or vice verse, due to start_external_replay signature change in #9788.

**Comment by trociny** [comment]
@dillaman #9788 merged, please rebase.

**Comment by dillaman** [comment]
@trociny pushed

File: src/tools/rbd_mirror/ImageReplayer.cc [hunk 1]
```diff
@@ -816,19 +816,44 @@ template <typename I>
...
-  flush(ctx);
+  ctx = new FunctionContext([this, ctx](int r) {
+      m_local_image_ctx->journal->stop_external_replay();
+      m_local_replay = nullptr;
+
+      if (r < 0) {
+        ctx->complete(r);
+        return;
+      }
+      m_local_image_ctx->journal->start_external_replay(&m_local_replay, ctx);
```
Comment (trociny) [inline src/tools/rbd_mirror/ImageReplayer.cc]: @dillaman start_external_replay is still missing on_close_request arg?

**Comment by dillaman** [comment]
@trociny forgot to append the commit

**Comment by trociny** [comment]
lgtm

**Pull request merged**